wordpress-mobile/WordPress-iOS
# Post types on posts and blog.

Adding support for persisting the `postType` of a `Post` entity as a `string`. Also adding new `PostType` entity for syncing the available list of post types for a site via `BlogService`.

It may seem apparent that the `Post` entity should actually set a relationship of a `PostType` entity instead of just the `string` for `postType`. However, this would prove burdensome for maintaining the list of `PostTypes` while trying to create and update `Posts` along the way. Since both APIs return a `string` for the name of the post type, it's best left to any observers of the `postType` string to lookup additional information within a `PostType`, as the API is designed itself.

**Note**: Originally I wanted to consolidate `AbstractPost` and `Page` into the single `Post` entity, but I discovered many different pieces of the code-base relying on a simple differentiation between a `Page` and `Post` entity. The deeper I got into the refactor, the more lost and the more code I had to touch, losing code clarity along the way. I still think that would be a great refactor, but a larger refactor would be required paying careful attention to particular code handling the differences in `Page` and `Post`.

**Testing Migration:**

1. Open the app on the develop branch.
2. Sync posts under the "Blog Posts" for a site.
3. While on the "Blog Posts" screen, Shfit+Cmd+H the app into the background to ensure the app state is saved and the posts are persisted to CD.
4. Switch to the `feature/post-types` branch.
5. Turn off Wifi or internet connection, so posts aren't automatically synced upon opening.
6. Build and run the app, the "Blog Posts" screen should appear.
7. Observe the posts displaying as previously left off.
8. Enable internet connection, pull to refresh.
9. Posts should display as normally in the next testing section.

**Testing:**

1. Open "Blog Posts" for a site.
2. Open the "Only Me" tab, posts by you should show.
3. Open the "Everyone" tab, posts by everyone should show.
4. Repeat steps for an XML-RPC connected site.

Needs review: @aerych
